hugo/content/docs/recipes/lexing/indentation-sensitive-languages.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Indentation-sensitive languages
+weight: 300
+---
+
+Some programming languages (such as Python, Haskell, and YAML) use indentation to denote nesting, as opposed to special non-whitespace tokens (such as `{` and `}` in C++/JavaScript).
+This can be difficult to express in the EBNF notation used for defining a language grammar in Langium, which is context-free.
+To achieve that, you can make use of synthetic tokens in the grammar which you would then redefine using Chevrotain in a custom token builder.
+
+Starting with Langium v3.2, such token builder (and an accompanying lexer) are provided for easy plugging into your language.
```

Comment:
Can you provide an explanation how this solution works. Other developers could benefit from those custom approach by imitating or adjusting the implementation.

Now it is just a blackbox. It is open source, yes, but it would be a good signal for others how to do things in Langium. Maybe a copy of the source code documentation is enough.

Reply:
I think it is easier to follow. Interested persons can lookup the source directly.